pkg/destroy/powervs/cloud-transit-gateways.go

```diff
@@ -0,0 +1,436 @@
+
+	for moreData {
+		// https://github.com/IBM/networking-go-sdk/blob/master/transitgatewayapisv1/transit_gateway_apis_v1.go#L184
+		gatewayCollection, response, err = o.tgClient.ListTransitGatewaysWithContext(ctx, listTransitGatewaysOptions)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to list transit gateways: %v and the respose is: %s", err, response)
+		}
+
+		for _, gateway = range gatewayCollection.TransitGateways {
+			if strings.Contains(*gateway.Name, o.InfraID) {
+				foundOne = true
+				o.Logger.Debugf("listTransitGateways: FOUND: %s, %s", *gateway.ID, *gateway.Name)
+				result = append(result, cloudResource{
+					key:      *gateway.ID,
+					name:     *gateway.Name,
+					status:   "",
+					typeName: transitGatewayTypeName,
+					id:       *gateway.ID,
+				})
+			}
+		}
+
+		if gatewayCollection.First != nil {
+			o.Logger.Debugf("listTransitGateways: First = %+v", *gatewayCollection.First.Href)
+		} else {
+			o.Logger.Debugf("listTransitGateways: First = nil")
+		}
+		if gatewayCollection.Limit != nil {
+			o.Logger.Debugf("listTransitGateways: Limit = %v", *gatewayCollection.Limit)
+		}
+		if gatewayCollection.Next != nil {
+			o.Logger.Debugf("listTransitGateways: Next = %+v", *gatewayCollection.Next)
+			listTransitGatewaysOptions.SetStart(*gatewayCollection.Next.Start)
+		} else {
+			o.Logger.Debugf("listTransitGateways: Next = nil")
+		}
+
+		moreData = gatewayCollection.Next != nil
+		o.Logger.Debugf("listTransitGateways: moreData = %v", moreData)
+	}
+	if !foundOne {
+		o.Logger.Debugf("listTransitGateways: NO matching transit gateway against: %s", o.InfraID)
+
+		listTransitGatewaysOptions = o.tgClient.NewListTransitGatewaysOptions()
+		listTransitGatewaysOptions.Limit = &perPage
+		moreData = true
+
+		for moreData {
+			gatewayCollection, response, err = o.tgClient.ListTransitGatewaysWithContext(ctx, listTransitGatewaysOptions)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to list transit gateways: %v and the respose is: %s", err, response)
+			}
+			for _, gateway = range gatewayCollection.TransitGateways {
+				o.Logger.Debugf("listTransitGateways: FOUND: %s, %s", *gateway.ID, *gateway.Name)
+			}
+			if gatewayCollection.First != nil {
+				o.Logger.Debugf("listTransitGateways: First = %+v", *gatewayCollection.First.Href)
+			} else {
+				o.Logger.Debugf("listTransitGateways: First = nil")
+			}
+			if gatewayCollection.Limit != nil {
+				o.Logger.Debugf("listTransitGateways: Limit = %v", *gatewayCollection.Limit)
+			}
+			if gatewayCollection.Next != nil {
+				o.Logger.Debugf("listTransitGateways: Next = %+v", *gatewayCollection.Next)
+				listTransitGatewaysOptions.SetStart(*gatewayCollection.Next.Start)
+			} else {
+				o.Logger.Debugf("listTransitGateways: Next = nil")
+			}
+			moreData = gatewayCollection.Next != nil
+			o.Logger.Debugf("listTransitGateways: moreData = %v", moreData)
+		}
+	}
+
+	return cloudResources{}.insert(result...), nil
```

Comment:
A bit confused why we're listing again in the Not Found case when we did so around L66 - L82.

Reply:
Like every other part of the destroy code, we only print out the matching TGs here:
https://github.com/hamzy/installer/blob/PowerVS-Add-transit-gateway-destroy/pkg/destroy/powervs/cloud-transit-gateways.go#L55
If no matching TGs were found, then we print out every TG here:
https://github.com/hamzy/installer/blob/PowerVS-Add-transit-gateway-destroy/pkg/destroy/powervs/cloud-transit-gateways.go#L97